binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowError(ExceptionFromError());
```

Comment:
Hard to tell. I don't know exactly what they are supposed to be used for, but I always assumed that type errors are for when an argument had the wrong type and range errors for when an argument value was outside some allowed range. Anything else would be generic errors.